### PR TITLE
framework/st_things: Arrange st_things's folder structure

### PR DESCRIPTION
**Pull request by heejin-kim**
Arrange and rename the framework/st_things files.
Also to avoid errors that can occur while building, arranged CFLAGS in Make.defs.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:1198: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/framework/things_data_manager.c:1198: ERROR: [IDT_M_TAB] please, no space before tabs
framework/src/st_things/things_stack/things_stack.c:37: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/things_stack.c:42: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/things_stack.c:44: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/things_stack.c:49: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/things_stack.c:51: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/utils/things_wait_handler.c [hunk 1]
```diff
@@ -21,14 +21,13 @@
 #include <stdlib.h>
 #include <pthread.h>
 
-#include "things_malloc.h"
-#include "things_logger.h"
-#include "things_wait_handler.h"
-
 #include "things_thread.h"
 #ifdef __ST_THINGS_RTOS__
```
Comment (jsdosa) [inline framework/src/st_things/things_stack/utils/things_wait_handler.c]: Do we need definition ?

Comment (heejin-kim) [reply framework/src/st_things/things_stack/utils/things_wait_handler.c]: Yes. It makes difference when create pthread.

**Comment by heejin-kim** [comment]
@juitem I checked dependencies and did test several times. For example, when I aligned include file  at things_data_manager.c, building binary is fine, but weird error occurs during easysetup. So changed the sequence of include files.

**Pull request closed without merging**